scalafmt-tests/src/test/resources/binPack/ParentConstructors.stat

```diff
@@ -114,3 +114,71 @@ def method[T <: ColType](): DTDataset[
     with FCol with GCol with HCol
     with ICol
 ]
+<<< #3091 always
+maxColumn = 100
+binPack.parentConstructors = always
+===
+object CoreReflection extends {
+    val u: ru.type = ru
+    val mirror: u.Mirror = u.runtimeMirror(classOf[scala.meta.Tree].getClassLoader)
+  }
+  with scala.meta.internal.trees.Reflection
+  with scala.meta.internal.tokens.Reflection
+>>>
+object CoreReflection extends {
+      val u: ru.type = ru
+      val mirror: u.Mirror = u.runtimeMirror(classOf[scala.meta.Tree].getClassLoader)
+    } with scala.meta.internal.trees.Reflection with scala.meta.internal.tokens.Reflection
```

Comment:
it only fixes `extends` position.
early stats and `} with ..` shouldn't be indented

Reply:
why shouldn't they be indented? everything between the beginning of a template and the opening brace of its body should be indented.

in your case,
```
... {
      val u: ru.type = ru
      val mirror: u.Mirror = u.runtimeMirror(classOf[scala.meta.Tree].getClassLoader)
    }
```
is not the body of `CoreReflection` but of some other (parent) construct. definitely has to be indented.

Reply:
here's another example, existing behaviour:

https://github.com/scalameta/scalafmt/blob/master/scalafmt-tests/src/test/resources/binPack/ParentConstructors.stat#L49

your proposal is contrary to that.

Reply:
If we treat that `parentConstructor` doesn't affect early init body and work only for `extends`  and with chain it wouldn't be contrary.

Let me try to fix it it properly, I didn't noticed that there is such setting at the moment when I was preparing a fix.

Reply:
can you explain how init body is different? other than indentation, what else would you like to do?